src/syscalls/mod.rs

```diff
@@ -417,6 +459,37 @@ pub unsafe extern "C" fn sys_write(fd: FileDescriptor, buf: *const u8, len: usiz
 	unsafe { write(fd, buf, len) }
 }
 
+#[hermit_macro::system]
+pub unsafe extern "C" fn sys_writev(fd: FileDescriptor, iov: *const iovec, iovcnt: i32) -> isize {
+	if iovcnt < 0 || iovcnt > IOV_MAX {
+		return (-crate::errno::EINVAL).try_into().unwrap();
+	}
+
+	let mut count: isize = 0;
+	let slice = unsafe { core::slice::from_raw_parts(iov, iovcnt.try_into().unwrap()) };
+
+	for i in slice {
```

Comment:
Nitpick: I'd suggest renaming `i` to `buf` and `slice` to `buffers`

Reply:
I fix it (see last commit)